### PR TITLE
Add ABI Fire Temperature, Day Convection, and Cloud Type composites.

### PR DESCRIPTION
**Pull request by sjoro**
This PR adds the ABI Fire Temperature, Day Convection and Cloud Type composites and enhancements. The recipes from which the RGBs have been constructed are documented in the `composites/abi.yaml`. 

Currently composites are named as:
`cira_fire_temperature`
`cira_day_convection`
`cimss_cloud_type`

Prefixes for the names should be removed if the recipes are considered to be "official".

 - [x] Fully documented

### CODE REVIEW

File: satpy/etc/enhancements/abi.yaml [hunk 1]
```diff
@@ -29,3 +31,41 @@ enhancements:
...
+
+  cira_day_convection:
+    standard_name: cira_day_convection
+    operations:
+      - name: stretch
+        method: !!python/name:satpy.enhancements.stretch
+        kwargs:
+          stretch: crude
+          min_stretch: [-35.0, -5.0, -0.75]
+          max_stretch: [5.0, 60.0, 0.25]
```
Comment (pnuu) [inline satpy/etc/enhancements/abi.yaml:57]: The B-channel stretch needs to be from `-75.0` to `25.0`, as Satpy handles reflectances as percentages.

Comment (pnuu) [reply satpy/etc/enhancements/abi.yaml:57]: Ah. The values you inserted were as the documentation stated, but the documentation has them wrong (mix of fraction and %-units). The current values are very very close to each other, so will practically only yield either 0 or 255 for the blue channel pixel values. I've used locally `-70` ... `20`, which gives nice images, so the values certainly need to be in that scale.

Comment (sjoro) [reply satpy/etc/enhancements/abi.yaml:57]: yes, i was doubting that stretch :), but obeyed the recipe. any thoughts on the AppVeyor fail on the PR?

Comment (pnuu) [reply satpy/etc/enhancements/abi.yaml:57]: That seems to be a decoding error when loading the ABI sensor composites. I'd hazard a guess that changing `μm` to `um` in the `fire_temperature` description might help.

Comment (pnuu) [reply satpy/etc/enhancements/abi.yaml:57]: Nevermind, the decoding error happens also on AHI compositor tests.

Comment (sjoro) [reply satpy/etc/enhancements/abi.yaml]: stretch fixed.

**Comment by djhoese** [comment]
Do any of these overlap with https://github.com/pytroll/satpy/pull/564

**Comment by sjoro** [comment]
@djhoese oh, ok. i only searched for "RGB" in open issues before making this PR and didn't find the #564. partial overlap, i have added the "cloud type" composite, which is not in there.

**Comment by mraspaud** [comment]
@sjoro feel like fixing the merge conflicts?

File: satpy/etc/composites/abi.yaml [hunk 2]
```diff
@@ -363,3 +363,315 @@ composites:
       - name: C04
       - name: C02
       - name: C05
+
```
Comment (mraspaud) [inline satpy/etc/composites/abi.yaml:366]:
```suggestion

```

**Pull request merged**